ni_measurementlink_service/session_management/_reservation.py

```diff
@@ -527,6 +715,42 @@ def initialize_session(
         """
         return self._initialize_session_core(session_constructor, instrument_type_id)
 
+    def initialize_multiplexer_session(
+        self,
+        session_constructor: Callable[[MultiplexerSessionInformation], TMultiplexerSession],
+        multiplexer_resource_name: Optional[str] = None,
+        multiplexer_type_id: Optional[str] = None,
+    ) -> ContextManager[TypedMultiplexerSessionInformation[TMultiplexerSession]]:
+        """Initialize a single multiplexer session.
+
+        This is a generic method that supports any multiplexer driver.
+
+        Args:
+            session_constructor: A function that constructs multiplexer sessions
+                based on multiplexer session information.
+
+            multiplexer_resource_name: Multiplexer resource name. If not
+                specified, the multiplexer resource name is ignored when
+                matching multiplexer sessions.
+
+            multiplexer_type_id: User-defined identifier for the multiplexer
+                type in the pin map editor. If not specified, the multiplexer
+                type id is ignored when matching multiplexer sessions.
+
+        Returns:
+            A context manager that yields a multiplexer session information
+                object. The session object is available via the ``session`` field.
+
+        Raises:
+            TypeError: If the argument types are incorrect.
+
+            ValueError: If no multiplexer sessions are available or
+                too many multiplexer sessions are available.
+        """
+        return self._initialize_multiplexer_session_core(
+            session_constructor, multiplexer_resource_name, multiplexer_type_id
+        )
+
     @requires_feature(SESSION_MANAGEMENT_2024Q1)
```

Comment:
@bkeryan I would like to know if we can remove these decorators.

Reply:
yes, claim https://github.com/ni/measurementlink-python/issues/516 when you do it